operator/duties/attester.go

```diff
@@ -172,27 +126,63 @@ func (h *AttesterHandler) processExecution(epoch phase0.Epoch, slot phase0.Slot)
 		return
 	}
 
-	if !h.network.PastAlanForkAtEpoch(h.network.Beacon.EstimatedEpochAtSlot(slot)) {
-		toExecute := make([]*genesisspectypes.Duty, 0, len(duties)*2)
-		for _, d := range duties {
-			if h.shouldExecute(d) {
-				toExecute = append(toExecute, h.toGenesisSpecDuty(d, genesisspectypes.BNRoleAttester))
-				toExecute = append(toExecute, h.toGenesisSpecDuty(d, genesisspectypes.BNRoleAggregator))
-			}
+	toExecute := make([]*genesisspectypes.Duty, 0, len(duties)*2)
+	for _, d := range duties {
+		if h.shouldExecute(d) {
+			toExecute = append(toExecute, h.toGenesisSpecDuty(d, genesisspectypes.BNRoleAttester))
+			toExecute = append(toExecute, h.toGenesisSpecDuty(d, genesisspectypes.BNRoleAggregator))
 		}
+	}
 
-		h.dutiesExecutor.ExecuteGenesisDuties(h.logger, toExecute)
-		return
+	h.dutiesExecutor.ExecuteGenesisDuties(h.logger, toExecute)
+}
+
+func (h *AttesterHandler) processIndicesChange(epoch phase0.Epoch, slot phase0.Slot) {
+	h.fetchCurrentEpoch = true
+
+	// reset next epoch duties if in appropriate slot range
+	if h.shouldFetchNexEpoch(slot) {
+		h.duties.Reset(epoch + 1)
+		h.fetchNextEpoch = true
 	}
 
-	toExecute := make([]*spectypes.ValidatorDuty, 0, len(duties))
-	for _, d := range duties {
-		if h.shouldExecute(d) {
-			toExecute = append(toExecute, h.toSpecDuty(d, spectypes.BNRoleAggregator))
+	h.indicesChanged = false
+}
+
+func (h *AttesterHandler) processReorg(ctx context.Context, epoch phase0.Epoch, reorgEvent ReorgEvent) {
+	// reset current epoch duties
+	if reorgEvent.Previous {
+		h.duties.Reset(epoch)
+		h.fetchCurrentEpoch = true
+		if h.shouldFetchNexEpoch(reorgEvent.Slot) {
+			h.duties.Reset(epoch + 1)
+			h.fetchNextEpoch = true
+		}
+
+		h.processFetching(ctx, epoch, reorgEvent.Slot)
+	} else if reorgEvent.Current {
+		// reset & re-fetch next epoch duties if in appropriate slot range,
+		// otherwise they will be fetched by the appropriate slot tick.
+		if h.shouldFetchNexEpoch(reorgEvent.Slot) {
+			h.duties.Reset(epoch + 1)
+			h.fetchNextEpoch = true
 		}
 	}
+}
 
-	h.dutiesExecutor.ExecuteDuties(h.logger, toExecute)
+func (h *AttesterHandler) processSlotTransition(epoch phase0.Epoch, slot phase0.Slot) {
+	slotsPerEpoch := h.network.Beacon.SlotsPerEpoch()
+
+	// If we have reached the mid-point of the epoch, fetch the duties for the next epoch in the next slot.
+	// This allows us to set them up at a time when the beacon node should be less busy.
+	if uint64(slot)%slotsPerEpoch == slotsPerEpoch/2-1 {
+		h.fetchNextEpoch = true
+	}
```

Comment:
Can use `if h.shouldFetchNexEpoch(...` here ? I guess applies to other Duties changed in this PR too.